src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -97,21 +112,93 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     toggle: togglePersonalisedSchedule,
   } = useShowHide(false);
 
+  const liveAndFutureEvents = useMemo(
+    () => relatedVenueEvents.filter(isEventLiveOrFuture),
+    [relatedVenueEvents]
+  );
+  const hasSavedEvents = liveAndFutureEvents
+    .map(
+      prepareForSchedule({
+        usersEvents: userEventIds,
+      })
+    )
+    .filter((event) => event.isSaved).length;
+
+  const isShowPersonalDownloadBtn = hasSavedEvents && showPersonalisedSchedule;
+  const liveEventsMinimalStartValue = Math.min(
+    ...liveAndFutureEvents.map((event) => event.start_utc_seconds)
+  );
+
+  const firstLiveEvent = liveAndFutureEvents.find(
+    (event) => event.start_utc_seconds === liveEventsMinimalStartValue
+  );
+
+  const minDate = useMemo(
+    () =>
+      firstLiveEvent
+        ? isEventLiveOrFuture(firstLiveEvent)
+          ? new Date(liveEventsMinimalStartValue).getTime()
+          : millisecondsToSeconds(todaysDateTime)
+        : minRangeValue,
+    [firstLiveEvent, liveEventsMinimalStartValue]
+  );
+
+  const firstRangeDayTimeValue = isDateLessOrEqualsToday({
+    dateValue: secondsToMilliseconds(minDate),
+    targetDateValue: millisecondsToSeconds(startOfToday().getTime()),
+  });
+
+  const firstRangeDateInSeconds = getUnixTime(
+    max([
+      new Date(secondsToMilliseconds(minDate)).getTime(),
+      todaysDate.getTime(),
+    ])
```

Comment:
still can't get why can't you just compare dates without all of these transforms date -> milisseconds -> date -> to milliseconds. Looks like a loooooot of unnecessary work. 
`max` should accept a date array, you don't have to parse everything to ms, so `minDate` (if it's date) and startOfToday() should be enough.

am I wrong?.. I still can't get why it looks so complicated. Maybe I'm losing something, please, explain..

Reply:
you're right, sorry
fixed in 60ff2eb